File: ALGORITHMS/C language/Searching/Linear_Search.c
// Linear search in c language

#include <stdio.h>

// Linear search function

int linear_search_function(int arr[], int target, int arr_length)
{
    // Iterate loop arr_length times.

    for (int i = 0; i < arr_length; i++)
    {
        // check whether target is present

        if (arr[i] == target)
        {
            // return index of target

            return i;
        }
    }
        return -1;
}

int main()
{
    int arr[] = {5, 8, 3, 7, 12, 3, 2, 10, 9};

    int arr_length = 9;

    int target = 121;

    // Linear search function call

    int ans = linear_search_function(arr, target, arr_length);
    if (ans == -1)
    {
        printf("Target is not present in Array.");
    }
    else
    {
        printf("\n Index of target id : %d", ans);
    }

    return 0;
}

File: ALGORITHMS/JAVA/Searching/Linear_Search.java
public class Linear_Search
{
  //Define a function for linear search
  
  public static void linearSearch(int arr[], int target)
  {
    //Iterating a loop to find out target
    
    for(int i=0;i<arr.length;i++)
    {
      if(arr[i] == target)
      {
        //Printing index at where index is found 
        
        System.out.println("The Target Is Found At Index : " + i);
        
        return;
      } 
    }  
    //If target is not found then printing the following massage
    
    System.out.println("The Target Is Not Found.");
  }
  public static void main(String args[])
  {
    int arr[]  = {12,5,8,3,24,3,19,33};
    
    linearSearch(arr, 24);
    
  }
}
